src/inmanta/agent/agent.py

```diff
@@ -539,13 +539,14 @@ def __init__(self, process: "Agent", name: str, uri: str, *, ensure_deploy_on_st
         self.ensure_deploy_on_start = ensure_deploy_on_start
 
         self.executor_manager: executor.ExecutorManager[executor.Executor] = process.executor_manager
+        self._executors_to_join: Optional[list[executor.Executor]] = None
 
     async def stop(self) -> None:
         self._stopped = True
         self._enabled = False
         self._disable_time_triggers()
         self._nq.cancel()
-        await self.executor_manager.stop_for_agent(self.name)
+        self._executors_to_join = await self.executor_manager.stop_for_agent(self.name)
```

Comment:
Why make it optional? It then requires the assertion later, while a list can just as well have 0 elements.